src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -1058,10 +1058,10 @@ private static ExpressionTokenKind MakeBestGuessOnNoSuffixStr(string numericStr,
                 decimal doubleToDecimalN;
 
                 // To keep the full presion of the current value, which if necessary is all 17 digits of precision supported by the Double type.
-                bool doubleCanBeDecimalR = decimal.TryParse(tmpDouble.ToString("R", CultureInfo.InvariantCulture), out doubleToDecimalR);
+                bool doubleCanBeDecimalR = decimal.TryParse(tmpDouble.ToString("R", CultureInfo.InvariantCulture), NumberStyles.Number, CultureInfo.InvariantCulture, out doubleToDecimalR);

```

Comment:
@lewischeng-ms  yes you are right, I changed to numberstyle to NumberStyles.Integer | NumberStyles.AllowDecimalPoint